atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepMetrics.java

```diff
@@ -60,16 +61,85 @@ void update(long value) {
         }
     }
 
-    void examinedCells(long numExamined) {
+    private class HistogramMetric {
+        private final Histogram histogram;
+
+        HistogramMetric(String name) {
+            this.histogram = metricsManager.getRegistry()
+                    .histogram(MetricRegistry.name(SweepMetrics.class, name));
+        }
+
+        void update(long value) {
+            this.histogram.update(value);
+        }
+
+    }
+
+    private class TableSpecificHistogramMetric {
+        private final String name;
+
+        TableSpecificHistogramMetric(String name) {
+            this.name = name;
+        }
+        void update(long value, TableReference tableRef) {
+            metricsManager.getTaggedRegistry().histogram(getTaggedMetric(name, tableRef)).update(value);
+        }
+    }
+
+    private class TableSpecificMeterMetric {
+        private final String name;
+
+        TableSpecificMeterMetric(String name) {
+            this.name = name;
+        }
+
+        void update(long value, TableReference tableRef) {
+            metricsManager.getTaggedRegistry().meter(getTaggedMetric(name, tableRef)).mark(value);
+        }
+    }
+
+    private MetricName getTaggedMetric(String name, TableReference tableRef) {
+        return MetricName.builder()
+                .safeName(MetricRegistry.name(SweepMetrics.class, name))
+                .safeTags(getTag(tableRef))
+                .build();
+    }
+
+    private Map<String,String> getTag(TableReference tableRef) {
+        // todo(gmaretic) Tag with table name once we can use tagged metrics
+        return ImmutableMap.of();
+    }
+
+    void examinedCellsOneIteration(long numExamined) {
```

Comment:
Why don't we tag the continuous metrics with the table name as well?

Reply:
We could, but not sure if it is valuable. This is very easy to add in the new impl.